lib/Onyx.js

```diff
@@ -855,21 +885,11 @@ function set(key, value) {
         Logger.logAlert(`Onyx.set() called after Onyx.merge() for key: ${key}. It is recommended to use set() or merge() not both.`);
     }
 
-    // If the value in the cache is the same as what we have then do not update subscribers unless they
-    // have initWithStoredValues: false then they MUST get all updates even if nothing has changed.
-    if (!cache.hasValueChanged(key, value)) {
-        cache.addToAccessedKeys(key);
-        notifySubscribersOnNextTick(key, value, subscriber => subscriber.initWithStoredValues === false);
-        return Promise.resolve();
-    }
-
-    // Adds the key to cache when it's not available
-    cache.set(key, value);
-    notifySubscribersOnNextTick(key, value);
+    // We don't wait until the new data is written to storage.
+    // Instead, we immediately notify subscribers and then return the new data
+    const setPromise = Storage.setItem(key, value);
 
-    // Write the thing to persistent storage, which will trigger a storage event for any other tabs open on this domain
-    return Storage.setItem(key, value)
-        .catch(error => evictStorageAndRetry(error, set, key, value));
+    return notify(key, value).then(() => setPromise).catch(error => evictStorageAndRetry(error, set, key, value));
```

Comment:
It seems unnecessary to have `notify` return a promise. We know `notify` always resolve immediately 

Why not just call `notify(key, value)` and leave the rest of the code unmodified?

```js
    notify(key, value, 'set');

    // Write the thing to persistent storage, which will trigger a storage event for any other tabs open on this domain
    return Storage.setItem(key, value)
        .catch(error => evictStorageAndRetry(error, set, key, value));
```